src/defi_integrations/MorphoScripts.sol

```diff
@@ -0,0 +1,261 @@
+     * @param shares The amount of shares to redeem
+     * @return assets The amount of assets redeemed
+     */
+    function redeem(address vault, uint256 shares) external returns (uint256) {
+        return IMetaMorpho(vault).redeem(shares, address(this), address(this));
+    }
+}
+
+contract MorphoBlueActions {
+    // To handle non-standard ERC20 tokens (i.e. USDT)
+    using SafeERC20 for IERC20;
+
+    error InvalidInput();
+
+    /**
+     * @notice Borrow assets or shares from a Morpho blue market on behalf of `onBehalf` and send assets to `receiver`
+     * @param morpho The address of the top level Morpho contract
+     * @param marketParams The market parameters of the individual morpho blue market to borrow assets from
+     * @param assets The amount of assets to borrow
+     * @param shares The amount of shares to borrow
+     * @param onBehalf The address that will own the increased borrow position
+     * @param receiver The address that will receive the borrowed assets
+     * @return assetsBorrowed The amount of assets borrowed
+     * @return sharesBorrowed The amount of shares minted
+     */
+    function borrow(
+        address morpho,
+        MarketParams memory marketParams,
+        uint256 assets,
+        uint256 shares,
+        address onBehalf,
+        address receiver
+    ) external returns (uint256, uint256) {
+        return IMorpho(morpho).borrow(marketParams, assets, shares, onBehalf, receiver);
+    }
+
+    /**
+     * @notice Repay assets or shares in a Morpho blue market on behalf of `onBehalf`
+     * @param morpho The address of the top level Morpho contract
+     * @param marketParams The market parameters of the individual morpho blue market
+     * @param assets The amount of assets to repay
+     * @param shares The amount of shares to repay
+     * @param onBehalf The address of the account to repay on behalf of
+     * @param data Arbitrary data to pass to the `onMorphoRepay` callback. Pass empty data if not needed
+     * @return assetsRepaid The amount of assets repaid
+     * @return sharesRepaid The amount of shares burned
+     */
+    function repay(
+        address morpho,
+        MarketParams memory marketParams,
+        uint256 assets,
+        uint256 shares,
+        address onBehalf,
+        bytes calldata data
+    ) external returns (uint256 assetsRepaid, uint256 sharesRepaid) {
+        if (assets > 0) {
+            IERC20(marketParams.loanToken).forceApprove(morpho, assets);
+            (assetsRepaid, sharesRepaid) = IMorpho(morpho).repay(marketParams, assets, shares, onBehalf, data);
+        } else if (shares > 0) {
+            IERC20(marketParams.loanToken).forceApprove(morpho, type(uint256).max);
+            (assetsRepaid, sharesRepaid) = IMorpho(morpho).repay(marketParams, assets, shares, onBehalf, data);
+            IERC20(marketParams.loanToken).forceApprove(morpho, 0);
+        } else {
+            revert InvalidInput();
```

Comment:
ok. Actually there was a moment I want to deprecated DefiScriptErrors haha. 
Since now its in different files, not really sure if we should group all those error in one random shared solidity file 🤔 
(But I guess that's for ease of parsing from client/backend side?)